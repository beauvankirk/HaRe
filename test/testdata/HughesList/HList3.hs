{-# LANGUAGE ScopedTypeVariables #-}
module Main where

--Super contrived example
explode :: Int -> [a] -> [a]
explode n lst = concat (map (\x -> replicate n x) lst)

f :: IO ()
f = do
  (lst :: [Int]) <- read <$> getLine
  let newLst = explode 3 lst
  print newLst

main = putStrLn "hello"
