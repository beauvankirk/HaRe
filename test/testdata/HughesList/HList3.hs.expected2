{-# LANGUAGE ScopedTypeVariables #-}
module HList3 where
import qualified Data.DList as DList
import Data.DList (DList)

--Super contrived example
explode :: Int -> [a] -> DList a
explode n lst = DList.concat (map (\x -> DList.replicate n x) lst)

f :: IO ()
f = do
  (lst :: [Int]) <- read <$> getLine
  let newLst = DList.toList (explode 3 lst)
  print newLst
