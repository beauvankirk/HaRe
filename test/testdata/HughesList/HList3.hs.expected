{-# LANGUAGE ScopedTypeVariables #-}
module HList3 where
import qualified HughesList.DList as DList
import HughesList.DList (DList)

--Super contrived example
explode :: Int -> DList a -> [a]
explode n lst = concat (map (\x -> replicate n x) (DList.toList lst))

f :: IO ()
f = do
  (lst :: [Int]) <- read <$> getLine
  let newLst = explode 3 (DList.fromList lst)
  print newLst
